dynawo/sources/Common/DYNTrace.cpp

```diff
@@ -148,19 +150,30 @@ void Trace::addAppenders_(const std::vector<TraceAppender>& appenders) {
   logging::attributes::current_thread_id::value_type currentId =
     logging::attributes::current_thread_id().get_value().extract<logging::attributes::current_thread_id::value_type>().get();
 
+  for (const TraceAppender& appender : appenders) {
+    if (appender.getTag() == Trace::variables()) {
+      variablesAppenderAndThreadId_ = std::make_pair(appender, currentId);
```

Comment:
I am still not sure why we need to store particular infos for this. Why can't we make a method called clearSink(Tag::variables) that would reset the sink (with the new method created) and then just recreate it from the infos it has at the start ? It would be more generic to any sink. Otherwise I would argue it would be better to just store appenders and not just one for the sake of generality.

Reply:
I believe it's not possible to export all the features from an added sink to re-create it later. The only way to recreate the sink is to store the info we used to create the sink at first and re-use the same info.

Reply:
And it's not possible de copy a sink

Reply:
ok so in this case I would prefer being more general and store infos about appenders during addAppenders_ method and put it in a map to be used later and on. And try to be the most generic to be reusable in future cases if need be. @rosiereflo is it ok with you or you prefer a specific method ?

Reply:
@gautierbureau I fully agree with your proposal